zio-http/shared/src/main/scala/zio/http/Status.scala

```diff
@@ -176,7 +176,7 @@ object Status {
     Try(code.toInt).toOption.map(fromInt)
 
   def fromInt(code: Int): Status = {
-    code match {
+    (code: @annotation.switch) match {
```

Comment:
Just for me: Is this an optional hint? Might the compiler create a switch without it? Might the compiler **not** create a switch with it?

Reply:
> Might the compiler create a switch without it

Yes
> Might the compiler not create a switch with it

No

This is very similar to `tailrec`; if the compiler can create a `lookupswitch` or `tableswitch`, it will do it even without the annotation. However, if the annotation is present, then it'll raise an error if a `lookupswitch` / `tableswitch` cannot be created from the expression.

Reply:
Thx! 👍